### PR TITLE
[REVIEW] Expose free and total memory in Python interface

### PR DESCRIPTION
**Pull request by gmarkall**
Whilst thinking about creating a memory manager plugin interface to use RMM with Numba, I noticed that the free and total memory doesn't seem to be available from the Python interface - this PR adds a wrapping of the `rmmGetInfo` function and exposes it as:

```python
rmm.get_info()
```

which will return, for example, `(16046292992, 16914055168)` (free and total memory in bytes).

This PR depends on PR #221, as the `test_rmm_getinfo_uninitialized` test causes a segfault without it.

### CODE REVIEW

File: python/rmm/_lib/lib.pyx [hunk 1]
```diff
@@ -269,3 +269,37 @@ def rmm_getallocationoffset(ptr, stream):
...
+    check_error(rmm_error)
+
+    return memory_pair(free, total)
+
+
+def rmm_getinfo(stream):
+    """Get total and free memory sizes using the RMM memory manager by calling
+    the librmm functions via Cython
+    """
+    cdef cudaStream_t c_stream = <cudaStream_t><size_t>stream
```
Comment (kkraus14) [inline python/rmm/_lib/lib.pyx]: This was recently discussed and we determined that a cudaStream is a pointer underneath and that it's more appropriate to use `uintptr_t` as compared to `size_t`.
```suggestion
    cdef cudaStream_t c_stream = <cudaStream_t><uintptr_t>stream
```

File: python/rmm/_lib/lib.pyx [hunk 2]
```diff
@@ -269,3 +269,37 @@ def rmm_getallocationoffset(ptr, stream):
...
+
+    cdef memory_pair memory_info = c_getinfo(
+        <cudaStream_t>c_stream
+    )
+
+    return int(memory_info.first), int(memory_info.second)
```
Comment (kkraus14) [inline python/rmm/_lib/lib.pyx]: I'm unsure what Numba expects for this interface here, but would it be possible / make sense to return a namedtuple here to make it more self-documenting?

File: python/rmm/_lib/lib.pyx (hunk 1 above)
Comment (gmarkall) [reply python/rmm/_lib/lib.pyx]: Thanks! I've made this change.

File: python/rmm/_lib/lib.pyx (hunk 2 above)
Comment (gmarkall) [reply python/rmm/_lib/lib.pyx]: There isn't really an expectation of an interface at the moment, but internally for a similar purpose Numba uses a `namedtuple` with `free` and `total` as members, so I've made `get_info` return a `namedtuple` with similar fields here.

**Comment by gmarkall** [comment]
Hopefully the comments are both addressed now, but I'd welcome any further feedback / suggestions - the build failures seem to be a result of one of the CI machines running out of disk space.

**Pull request merged**